src/FieldDataPluginFramework.Serialization/JsonConfig.cs

```diff
@@ -290,9 +291,15 @@ private static void ConfigureSpecialParsers()
                 json.Get<DateTimeOffset>(nameof(GageHeightMeasurement.MeasurementTime)),
                 json.Get<bool>(nameof(GageHeightMeasurement.Include))));
 
-            Configure(json => json.HasProperty(nameof(ManualGaugingDischargeSection.VelocityObservationMethod))
-                ? (ChannelMeasurementBase)json.JsonText.FromJson<ManualGaugingDischargeSection>()
-                : json.JsonText.FromJson<AdcpDischargeSection>());
+            Configure(json => json.HasProperty(nameof(ManualGaugingDischargeSection.StartPoint))
```

Comment:
This is the nasty deserialization expression that figures out what type of channel measurement to deserialize based on certain properties it discovers in the JSON. If the `ChannelMeasurementBase` class had some sort of `ChannelMeasurementType` property, the logic would be cleaner.

Instead, I can only cobble together a 5-layer if-then-else based on properties that only exist in only one of the types. I had to muck around to find properties that can never be null (and therefore might be omitted by JSON serialization logic).

| Type | Expected property |
|---|---|
| ManualGauging | `StartPoint` |
| ADCP | `AdcpDeviceType` |
| EngineeredStructure | `StructureEquation` (this can never be null, right?) |
| Volumetric | `MeasurementContainerUnit` |
| Other | I don't think there are any required-and-unique-to-Other properties, so this is the "else" condition. |